packages/model-viewer/src/features/scene-graph/texture-info.ts

```diff
@@ -146,15 +146,15 @@ export class TextureInfo extends ThreeDOMElement implements
     );
   }
 
-  get texture(): Texture|null {
+  get texture(): Texture {
     return this[$texture];
   }
 
   setTexture(texture: Texture|null): void {
     const threeTexture: ThreeTexture|null =
         texture != null ? texture.source[$underlyingTexture] : null;
     let encoding: TextureEncoding = sRGBEncoding;
-    this[$texture] = texture;
+    this[$texture] = texture ?? new Texture(this);
```

Comment:
@timmmeh I added this because I wanted to make `.texture` always defined. Looking at the constructor, this seemed to be what you did when nothing was specified. Does this seem reasonable?

Reply:
So I added nullness support so you could remove a texture, basically allows for an empty texture-info.  I think this change would break that, but you can check it with the "Create Textures" example you should be able to add a normal map to the duck and remove it.

Reply:
Hmm, I guess I was hoping this would still null it out. I'll check. If it breaks that example, we should probably add a test for this.

Reply:
Yeah, the example seems to work fine switching back to none. Can you review again and make sure you're happy with the logic here?

Reply:
Is there a new commit for this? I don't see it yet.
So the other issue that could crop up here is on the API side, if the texture is not null or at least indicated to the user to be invalid than you could easily call setURI() (like in the Swap Textures example)
```
 let applyPBRTexture = (channel, event) => {
    material.pbrMetallicRoughness[channel].texture.source.setURI(event.target.value);
  }
```
If texture were null you could check use that to at least check the if the texture is valid, if not I think it would succeed but just not do anything, not 100% sure it might still produce an error somewhere, but the point is there is no check to see if it's actually a valid texture or not.

Reply:
Interesting; so how is this different than how your constructor works when there is no texture in the glTF? I thought I was following that same logic as that, but I may be mistaken.

Reply:
I may have not completed that idea thoroughly in the code, although that was my intent at one point and I think it makes sense for the API, taking a look...

Reply:
So taking a closer look, your right the constructor is creating a valid texture, in the initial design of things this was how it was working but once we switched to the setTexture() API I was going to change this but forgot. I'd still like to do that though, so I'd say I can make that change now, I think it will be a small change

Reply:
Ok fixed the API to create a null Texture object when the backing three-texture is null or undefined. Also added test to exercise this code.
https://github.com/google/model-viewer/pull/2568

Reply:
Thanks!

Reply:
Fixed.